File: app/src/main/java/com/ephirium/storyline/feature/ui/PostsPageFragment.java
package com.ephirium.storyline.feature.ui;

import static com.ephirium.storyline.StorylineLog.error;
import static com.ephirium.storyline.StorylineLog.logError;

import android.app.Activity;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.lifecycle.ViewModelProvider;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.ephirium.storyline.databinding.FragmentPostsPageBinding;
import com.ephirium.storyline.feature.model.Post;
import com.ephirium.storyline.feature.presentation.PostsPageViewModel;
import com.ephirium.storyline.feature.ui.recycler.PostsAdapter;
import com.ephirium.storyline.feature.ui.recycler.PostsCallback;
import com.ephirium.storyline.feature.ui.recycler.PostsCallbackBuilder;

import org.jetbrains.annotations.Contract;

public class PostsPageFragment extends Fragment {

    private FragmentPostsPageBinding binding;

    private PostsPageViewModel viewModel;

    private final PostsAdapter adapter = new PostsAdapter(new PostsCallbackBuilder()
            .addOnClickCallback(post -> {

            })
            .addOnLikeCallback(post -> {

            }).build());

    public PostsPageFragment() {
    }

    @NonNull
    public static PostsPageFragment newInstance() {
        return new PostsPageFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentPostsPageBinding.inflate(getLayoutInflater());
        binding.recycler.setAdapter(adapter);

        if (getActivity() != null) {
            viewModel = new ViewModelProvider(getActivity()).get(PostsPageViewModel.class);
            viewModel.postsList.observe(getActivity(), adapter::setPosts);
        }
        else{
            error("Activity does not exist", PostsPageFragment.class);
            throw new RuntimeException("Activity does not exist");
        }
        viewModel.observePosts();

        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/ephirium/storyline/ui/fragment/PostsPageFragment.kt
package com.ephirium.storyline.ui.fragment

import by.kirich1409.viewbindingdelegate.viewBinding
import com.ephirium.storyline.R

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import android.view.View;
import com.ephirium.common.log.log

import com.ephirium.storyline.databinding.FragmentPostsPageBinding;
import com.ephirium.storyline.presentation.PostsPageViewModel;
import com.ephirium.storyline.ui.recycler.post.PostAdapter;
import com.ephirium.storyline.ui.recycler.post.callback.PostCallback;

class PostsPageFragment : Fragment(R.layout.fragment_posts_page) {

    private val binding: FragmentPostsPageBinding by viewBinding()

    private val viewModel: PostsPageViewModel by lazy { ViewModelProvider(requireActivity())[PostsPageViewModel::class.java] }

    private val adapter = PostAdapter(PostCallback {

    }.addOnMoveCallback { from, to ->
        run {

        }
    }.addOnSwipeCallback { direction, position ->
        run {

        }
    })

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.recycler.adapter = adapter

        log("PostPageFragment")

        viewModel.postsList.observe(requireActivity(), adapter::setPosts)

        if (savedInstanceState != null) viewModel.observePosts()
    }
}